PactNet/Mocks/MockHttpService/MockProviderService.cs

```diff
@@ -125,32 +127,48 @@ public void VerifyInteractions()
             Async.RunSync(() => _adminHttpClient.SendAdminHttpRequest(HttpVerb.Get, $"{Constants.InteractionsVerificationPath}?example_description={testContext}"));
         }
 
-        public void SendAdminHttpRequest(HttpVerb method, string path)
+        public void SendAdminHttpRequest(HttpVerb method, string path, Dictionary<string, string> headers = null)
         {
-            Async.RunSync(() => _adminHttpClient.SendAdminHttpRequest(method, path));
+            Async.RunSync(() => _adminHttpClient.SendAdminHttpRequest(method, path, headers:headers));
         }
 
         public void Start()
         {
-            StopRunningHost();
+            if (UseRemoteMockService == false)
+            {
+                StopRunningHost();
+                _host = _hostFactory(BaseUri);
+                _host.Start();
+            }
 
-            _host = _hostFactory(BaseUri);
-            _host.Start();
         }
 
         public void Stop()
         {
             ClearAllState();
-            StopRunningHost();
+            if (UseRemoteMockService == false)
+            {
+                StopRunningHost();
+            }
         }
 
         public void ClearInteractions()
         {
-            if (_host != null)
+            if (_host != null && UseRemoteMockService == false)
             {
                 var testContext = BuildTestContext();
                 Async.RunSync(() => _adminHttpClient.SendAdminHttpRequest(HttpVerb.Delete, $"{Constants.InteractionsPath}?example_description={testContext}"));
             }
+            else
+            {
+                ClearAllInteractions();
+            }
+
+        }
+
+        public void ClearAllInteractions()
+        {
+            Async.RunSync(() => _adminHttpClient.SendAdminHttpRequest(HttpVerb.Delete, $"{Constants.InteractionsPath}?"));
         }

```

Comment:
I have one big suggestion to consider for changes in this class.

Instead of setting additional property in this quite big class, Maybe you could consider creating yet another instance of IMockProviderService and extend a bit concept of `_mockProviderServiceFactory` in PactBuilder so it will not be just Func, but full citizen class? This class based on `useRemoteMockService` param will decide, which realization provide. BTW I do not see why `MockProviderService` is not internal, IMHO it should be.

You can even use some strategy or template method pattern. For instance you will have base abstract class with a few abstract methods related to: Start, Stop, Clear (mostly current implementation of MockProviderService) and two non abstract e.g. AttachedMockProviderService and StandAloneMockProviderService, with abstract method implementations. Hope it is clear.

Reply:
That's a very good suggestion! I am definitely taking good note of this. 

I propose to do this refactor in an upcoming separate pull request (as the current one is already big enough IMO)